cmd/clioptions/metrics.go

```diff
@@ -137,40 +138,94 @@ func (m metricsTimer) Record(duration time.Duration) {
 	m.prom.Observe(duration.Seconds())
 }
 
+// processCollector is a cross-platform Prometheus collector that uses gopsutil
+// to collect CPU and memory metrics. The standard ProcessCollector from Prometheus
+// only works on Linux (requires /proc).
+type processCollector struct {
+	pid     int32
+	cpuDesc *prometheus.Desc
+	memDesc *prometheus.Desc
+}
+
+func newProcessCollector() *processCollector {
+	return &processCollector{
+		pid: int32(os.Getpid()),
+		cpuDesc: prometheus.NewDesc(
+			"process_cpu_seconds_total",
+			"Total user and system CPU time spent in seconds.",
+			nil, nil,
+		),
+		memDesc: prometheus.NewDesc(
+			"process_resident_memory_bytes",
+			"Resident memory size in bytes.",
+			nil, nil,
+		),
+	}
+}
+
+func (c *processCollector) Describe(ch chan<- *prometheus.Desc) {
+	ch <- c.cpuDesc
+	ch <- c.memDesc
+}
+
+func (c *processCollector) Collect(ch chan<- prometheus.Metric) {
+	p, err := process.NewProcess(c.pid)
+	if err != nil {
+		return
+	}
+
+	// CPU times (user + system)
+	if times, err := p.Times(); err == nil {
+		ch <- prometheus.MustNewConstMetric(c.cpuDesc, prometheus.CounterValue, times.User+times.System)
```

Comment:
You don't want to use these times, you'll want to use https://pkg.go.dev/github.com/shirou/gopsutil/v4@v4.25.10/cpu#Percent which will be much more interpretable